components/google-cloud/google_cloud_pipeline_components/experimental/forecasting/prepare_data_for_train/component.py

```diff
@@ -0,0 +1,169 @@
+  time_series_attribute_columns = []
+  available_at_forecast_columns = []
+  unavailable_at_forecast_columns = []
+  column_transformations = []
+  predefined_split_column = (
+      '' if 'predefined_splits_column' not in primary_metadata
+      else primary_metadata['predefined_splits_column'])
+  weight_column = (
+      '' if 'weight_column' not in primary_metadata
+      else primary_metadata['weight_column'])
+
+  for name, details in column_metadata.items():
+    if name == predefined_split_column or name == weight_column:
+      continue
+    # Determine temporal type of the column
+    if details['tag'] == 'vertex_time_series_key':
+      time_series_identifier_column = name
+      continue
+    elif details['tag'] == 'primary_table':
+      if name in (
+          primary_metadata['time_series_identifier_columns']
+      ):
+        time_series_attribute_columns.append(name)
+      elif name == primary_metadata['target_column']:
+        unavailable_at_forecast_columns.append(name)
+      elif name in primary_metadata['unavailable_at_forecast_columns']:
+        unavailable_at_forecast_columns.append(name)
+      else:
+        available_at_forecast_columns.append(name)
+    elif details['tag'] == 'attribute_table':
+      time_series_attribute_columns.append(name)
+    elif details['tag'] == 'fe_static':
+      time_series_attribute_columns.append(name)
+    elif details['tag'] == 'fe_available_past_future':
+      available_at_forecast_columns.append(name)
+    elif details['tag'] == 'fe_available_past_only':
+      unavailable_at_forecast_columns.append(name)
+
+    # Determine data type of the column. Note: columns without transformations
+    # (except for target) will be ignored during training.
+    if feature_columns and name not in feature_columns:
+      continue
+    if details['type'] == 'STRING':
+      trans = {'categorical': {'column_name': name}}
+      column_transformations.append(trans)
+    elif details['type'] in ['INT64', 'INTEGER', 'NUMERIC', 'FLOAT64', 'FLOAT']:
+      trans = {'numeric': {'column_name': name}}
+      column_transformations.append(trans)
+    elif details['type'] in ['DATETIME', 'DATE', 'TIMESTAMP']:
+      trans = {'timestamp': {'column_name': name}}
+      column_transformations.append(trans)
+
+  return (
+      time_series_identifier_column,
+      json.dumps(time_series_attribute_columns),
+      json.dumps(available_at_forecast_columns),
+      json.dumps(unavailable_at_forecast_columns),
+      json.dumps(column_transformations),
+      bigquery_table_uri,
+      primary_metadata['target_column'],
+      primary_metadata['time_column'],
+      predefined_split_column,
+      weight_column,
+      primary_metadata['time_granularity']['unit'].lower(),
+      str(primary_metadata['time_granularity']['quantity']),
+  )
+
+
+if __name__ == '__main__':
+  prepare_data_for_train_op = create_component_from_func(
+      prepare_data_for_train,
+      base_image='python:3.8',
```

Comment:
This part seems to be redundant since you are loading the component in` components/google-cloud/google_cloud_pipeline_components/experimental/forecasting/__init__.py` is there a specific reason for using __main__ and loading the method this way ?

Reply:
I was using the main method myself to generate the yaml file. I thought this is the recommended way, as I found an example in experimental/Optimizer/Suggest_trials/component.py

Reply:
yeah this is a python lightweight component but we still want to publish a YAML for it. this code should allow a yaml to be generated. not sure if there is any better way either.

Reply:
sgtm